backend/controllers/appointmentConfirmDeletion.js

```diff
@@ -0,0 +1,86 @@
+const packAppointmentDetails = (req, details) => {
+  req.flash('appointmentDetails', details)
+}
+
+const unpackAppointmentDetails = req => {
+  const appointmentDetails = req.flash('appointmentDetails')
+  if (!appointmentDetails || !appointmentDetails.length) throw new Error('Appointment details are missing')
+
+  return appointmentDetails.reduce(
+    (acc, current) => ({
+      ...acc,
+      ...current,
+    }),
+    {}
+  )
+}
+
+const clearAppointmentDetails = req => {
+  req.flash('appointmentDetails')
+}
+
+module.exports = ({ whereaboutsApi }) => {
+  const renderTemplate = (req, res, errors) => {
+    const appointmentDetails = unpackAppointmentDetails(req)
+    const {
+      id,
+      isRecurring,
+      additionalDetails,
+      basicDetails,
+      prepostData,
+      recurringDetails,
+      timeDetails,
+    } = appointmentDetails
+
+    // Save for possible re-showing on error
+    packAppointmentDetails(req, appointmentDetails)
```

Comment:
Not fully sure if this is needed. Wouldn't you just create new errors within the post request then redirect back to here?

Reply:
I do - but the flash is reset just above this, so without this there would be an error but no data in the flash